lib/Service/SnoozeService.php

```diff
@@ -287,12 +289,25 @@ private function wakeMessagesByAccount(Account $account): void {
 		try {
 			$messageIdsToDelete = [];
 			foreach ($messages as $message) {
+				$srcMailboxId = $this->messageSnoozeMapper->getSrcMailboxId($message->getMessageId());
```

Comment:
The additional roundtrip for srcMailboxId is a bit sad.

Idea 1: Introduce SnoozedMessage and SnoozedMessageMapper. SnoozedMessage should extend Message. SnoozedMessageMapper should extend QBMapper. That's nice, but also a bit weird because SnoozedMessageMapper inherits a couple of methods from QBMapper which we need to adjust. 

Idea 2: Don't use a DTO at all. For the unsnooze operation, we need the srcMailboxId and message uid. I think it's okay to change findMessagesToUnSnooze to return an array. I think that's much simpler than idea 1. 

If you do not have time, I'm happy to continue with the current iteration.

Reply:
:+1: